libs/model/src/community/CreateStakeTransaction.command.ts

```diff
@@ -0,0 +1,153 @@
+import type { Command } from '@hicommonwealth/core';
+import { commonProtocol, schemas } from '@hicommonwealth/core';
+import { Op } from 'sequelize';
+import Web3 from 'web3';
+import { models } from '../database';
+import { mustExist } from '../middleware/guards';
+import { StakeTransactionInstance } from '../models/stake_transaction';
+
+async function createStake(
+  web3: Web3,
+  transactionHash: string,
+  communityId: string,
+  namespace: string,
+  communityStakeAddress: string,
+) {
+  const [transaction, txReceipt] = await Promise.all([
+    web3.eth.getTransaction(transactionHash),
+    web3.eth.getTransactionReceipt(transactionHash),
+  ]);
+  const timestamp: number = (
+    await web3.eth.getBlock(transaction.blockHash as string)
+  ).timestamp as number;
+
+  if (![transaction.from, transaction.to].includes(communityStakeAddress)) {
+    throw new Error(
+      'This transaction is not associated with a community stake',
+    );
+  }
+
+  const { 0: stakeId, 1: value } = web3.eth.abi.decodeParameters(
+    ['uint256', 'uint256'],
+    txReceipt.logs[0].data,
+  );
+
+  const callData = {
+    to: txReceipt.logs[0].address, // src of Transfer single
+    data: '06fdde03', // name function selector
+  };
+
+  const response = await web3.eth.call(callData);
+  const name = web3.eth.abi.decodeParameter(
+    'string',
+    response,
+  ) as unknown as string;
+  if (name !== namespace) {
+    throw new Error('Transaction is not associated with provided community');
+  }
+
+  const {
+    0: trader,
+    // 1: namespace,
+    2: isBuy,
+    // 3: communityTokenAmount,
+    4: ethAmount,
+    // 5: protocolEthAmount,
+    // 6: nameSpaceEthAmount,
+  } = web3.eth.abi.decodeParameters(
+    ['address', 'address', 'bool', 'uint256', 'uint256', 'uint256', 'uint256'],
+    txReceipt.logs[1].data,
+  );
+
+  const stakeAggregate = await models.StakeTransaction.create({
+    transaction_hash: transactionHash,
+    community_id: communityId,
+    stake_id: parseInt(stakeId),
+    stake_amount: parseInt(value),
+    stake_price: ethAmount,
+    address: trader,
+    stake_direction: isBuy ? 'buy' : 'sell',
+    timestamp: timestamp,
+  });
+
+  return stakeAggregate?.get({ plain: true });
```

Comment:
Why not return `stakeAggregate.toJSON()` so we can avoid an unnecessary round-trip to the DB?

Reply:
Get also does not make a db call. It is just a wrapper around getting the data values. I can convert to toJSON but it essentially does the same thing.